lightning/src/ln/channelmanager.rs

```diff
@@ -1451,6 +1462,14 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 				if let Some(short_id) = chan.get().get_short_channel_id() {
 					channel_state.short_to_id.remove(&short_id);
 				}
+				let mut pending_events_lock = self.pending_events.lock().unwrap();
+				if peer_node_id.is_some() {
+					if let Some(peer_msg) = peer_msg {
+						pending_events_lock.push(events::Event::ChannelClosed { channel_id: *channel_id, reason: ClosureReason::CounterpartyForceClosed { peer_msg: peer_msg.to_string() } });
```

Comment:
I still think it would be much clearer to combine `peer_msg` and `peer_node_id` into one variable, making this code more readable and making it clear that this is infallible.

Reply:
Unless you think this is really required to have the feature functional, feel free to take it as a follow-up :)

Reply:
Sure, gotta open an issue and do it, though :)